src/FulfilledPromise.php

```diff
@@ -57,6 +57,9 @@ public function otherwise(callable $onRejected)
 
     public function wait($unwrap = true, $defaultDelivery = null)
     {
+        $queue = Utils::queue();
+        $queue->run();
```

Comment:
Another possible solution is to remember the newly created promise (`$p = new Promise([$queue, 'run']);`) and wait for `$p` here.